### PR TITLE
Add subsequent/alternate registration schemas

### PR DESCRIPTION
**Pull request by skenaja**
https://opencorporates.atlassian.net/browse/OCD-994

Also see https://github.com/openc/openc/pull/2277.

### CODE REVIEW

File: schemas/includes/related-registration.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "properties": {
+    "registration_type": {
+      "description": "The type of the related registration. Use 'predecessor' to link to the prior registration, 'successor' to link to the replacement registration or 'alternate' to link to an alternative registration, e.g. for companies registered more than once in a jurisdiction.",
+      "enum": [
+        "predecessor",
+        "successor",
+        "alternate"
+      ]
+    },
+    "related_entity": {
```
Comment (dominicsayers) [inline schemas/includes/related-registration.json]: Does it need to be an entity rather than a company?

Comment (skenaja) [reply schemas/includes/related-registration.json]: Probably not. Will push change

**Comment by skenaja** [comment]
rebased against master

File: schemas/alternate-registration-schema.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      "type": "string",
+      "format": "date"
+    },
+    "alternate_registration_end_date": {
+      "description": "date when alternate registration ended",
+      "type": "string",
+      "format": "date"
+    },
+    "publication_date": {
+      "description": "TBC",
```
Comment (skenaja) [inline schemas/alternate-registration-schema.json]: "the publication date of the filing/notice that gives details of the alternate registration"

File: schemas/alternate-registration-schema.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  "description": "A representation of multiple registrations for the same entity",
+  "type": "object",
+  "properties": {
+    "data_type": {
+      "enum": [
+        "alternate-registration"
+      ]
+    },
+    "alternate_entities": {
+      "description": "TBC",
```
Comment (skenaja) [inline schemas/alternate-registration-schema.json]: "array of entities that are linked via alternate registrations"

**Comment by skenaja** [comment]
Suggested text for "TBC" listed above but otherwise 👍

File: schemas/alternate-registration-schema.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  ],
+  "anyOf": [
+    {
+      "required": [
+        "start_date"
+      ]
+    },
+    {
+      "required": [
+        "end_date"
```
Comment (bensymonds) [inline schemas/alternate-registration-schema.json]: Just noticed we still have this, but no `end_date` property! Odd that json-schema doesn't complain.

File: schemas/alternate-registration-schema.json (hunk 2 above)
Comment (bensymonds) [reply schemas/alternate-registration-schema.json]: ✅

File: schemas/alternate-registration-schema.json (hunk 3 above)
Comment (bensymonds) [reply schemas/alternate-registration-schema.json]: ✅

**Comment by bensymonds** [comment]
Thanks both. I think this is good to go now :+1: Will merge and update things shortly.

**Pull request merged**